runelite-client/src/main/java/net/runelite/client/plugins/keyremapping/KeyRemappingListener.java

```diff
@@ -0,0 +1,314 @@
+					e.setKeyCode(KeyEvent.VK_F2);
+				}
+				else if (three.matches(e))
+				{
+					modified.put(e.getKeyCode(), KeyEvent.VK_F3);
+					e.setKeyCode(KeyEvent.VK_F3);
+				}
+				else if (four.matches(e))
+				{
+					modified.put(e.getKeyCode(), KeyEvent.VK_F4);
+					e.setKeyCode(KeyEvent.VK_F4);
+				}
+				else if (five.matches(e))
+				{
+					modified.put(e.getKeyCode(), KeyEvent.VK_F5);
+					e.setKeyCode(KeyEvent.VK_F5);
+				}
+				else if (six.matches(e))
+				{
+					modified.put(e.getKeyCode(), KeyEvent.VK_F6);
+					e.setKeyCode(KeyEvent.VK_F6);
+				}
+				else if (seven.matches(e))
+				{
+					modified.put(e.getKeyCode(), KeyEvent.VK_F7);
+					e.setKeyCode(KeyEvent.VK_F7);
+				}
+				else if (eight.matches(e))
+				{
+					modified.put(e.getKeyCode(), KeyEvent.VK_F8);
+					e.setKeyCode(KeyEvent.VK_F8);
+				}
+				else if (nine.matches(e))
+				{
+					modified.put(e.getKeyCode(), KeyEvent.VK_F9);
+					e.setKeyCode(KeyEvent.VK_F9);
+				}
+				else if (ten.matches(e))
+				{
+					modified.put(e.getKeyCode(), KeyEvent.VK_F10);
+					e.setKeyCode(KeyEvent.VK_F10);
+				}
+				else if (eleven.matches(e))
+				{
+					modified.put(e.getKeyCode(), KeyEvent.VK_F11);
+					e.setKeyCode(KeyEvent.VK_F11);
+				}
+				else if (twelve.matches(e))
+				{
+					modified.put(e.getKeyCode(), KeyEvent.VK_F12);
+					e.setKeyCode(KeyEvent.VK_F12);
+				}
+			}
+			
+			switch (e.getKeyCode())
+			{
+				case KeyEvent.VK_ENTER:
+				case KeyEvent.VK_SLASH:
+				case KeyEvent.VK_COLON:
+					// refocus chatbox
+					plugin.setTyping(true);
+					clientThread.invoke(() ->
+					{
+						plugin.unlockChat();
+					});
+					break;
+			}
+			
+		}
+		else
+		{
+			switch (e.getKeyCode())
+			{
+				case KeyEvent.VK_ENTER:
+					plugin.setTyping(false);
+					clientThread.invoke(() ->
+					{
+						plugin.lockChat();
+					});
+					break;
+				case KeyEvent.VK_ESCAPE:
+					plugin.setTyping(false);
+					clientThread.invoke(() ->
+					{
+						client.setVar(VarClientStr.CHATBOX_TYPED_TEXT, "");
+						plugin.lockChat();
+					});
+					break;
+				case KeyEvent.VK_BACK_SPACE:
+					if (Strings.isNullOrEmpty(client.getVar(VarClientStr.CHATBOX_TYPED_TEXT)))
+					{
+						plugin.setTyping(false);
+						clientThread.invoke(() -> plugin.lockChat());
+					}
+			}
+		}
+	}
+
+	@Override
+	public void keyReleased(KeyEvent e)
+	{
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			return;
+		}
+
+		if (plugin.chatboxFocused() && !plugin.isTyping())
+		{
+			modified.remove(e.getKeyCode());
+
+			if (config.wasdCameraRemap())
+			{
+				if (config.up().matches(e))
+				{
+					e.setKeyCode(KeyEvent.VK_UP);
+				}
+				else if (config.down().matches(e))
+				{
+					e.setKeyCode(KeyEvent.VK_DOWN);
+				}
+				else if (config.left().matches(e))
+				{
+					e.setKeyCode(KeyEvent.VK_LEFT);
+				}
+				else if (config.right().matches(e))
+				{
+					e.setKeyCode(KeyEvent.VK_RIGHT);
+				}
+				
+				
+				if (config.functionKeyRemap())
```

Comment:
why is this inside of the `if (config.wasdCameraRemap())`?

Reply:
fucked it up after rebasing